### DISCUSSION TITLE
Blazor webassembly and webgl

### DISCUSSION THREAD

**Discussion by unicomp21** [dotnet/AspNetCore.Docs#24458] [Ideas]
Anyone have thoughts on providing webgl/webassembly/dllimport bindings for google filament?

Being able to do webgl programming from c# would be pretty cool.
https://github.com/google/filament

The thinking would be to leverage the codegen for typescript bindings to support c# too.
https://github.com/google/filament/blob/0229077293f0dda532d8b4989b3f5f773eccf6b6/web/filament-js/jsbindings.cpp

**Comment by unicomp21** [comment]
Lol, given Skia has already been done, why not filament too?

**Comment by rishit-singh** [comment]
I second this. We need WebGL bindings for Blazor. Until then maybe it can be achieved by calling it via IJSRuntime.
